common/src/main/java/io/druid/math/expr/Function.java

```diff
@@ -40,761 +36,7 @@
-    {
-      if (args.size() != 3) {
-        throw new IAE("function 'if' needs 3 arguments");
-      }
-
-      ExprEval x = args.get(0).eval(bindings);
-      return x.asBoolean() ? args.get(1).eval(bindings) : args.get(2).eval(bindings);
-    }
-  }
-
-  class CastFunc extends DoubleParam
-  {
-    @Override
-    public String name()
-    {
-      return "cast";
-    }
-
-    @Override
-    protected ExprEval eval(ExprEval x, ExprEval y)
-    {
-      ExprType castTo;
-      try {
-        castTo = ExprType.valueOf(y.asString().toUpperCase());
-      }
-      catch (IllegalArgumentException e) {
-        throw new IAE("invalid type '%s'", y.asString());
-      }
-      return x.castTo(castTo);
-    }
-  }
-
-  class TimestampFromEpochFunc implements Function
-  {
-    @Override
-    public String name()
-    {
-      return "timestamp";
-    }
-
-    @Override
-    public ExprEval apply(List<Expr> args, Expr.ObjectBinding bindings)
-    {
-      if (args.size() != 1 && args.size() != 2) {
-        throw new IAE("function '%s' needs 1 or 2 arguments", name());
-      }
-      ExprEval value = args.get(0).eval(bindings);
-      if (value.type() != ExprType.STRING) {
-        throw new IAE("first argument should be string type but got %s type", value.type());
-      }
-
-      DateTimeFormatter formatter = ISODateTimeFormat.dateOptionalTimeParser();
-      if (args.size() > 1) {
-        ExprEval format = args.get(1).eval(bindings);
-        if (format.type() != ExprType.STRING) {
-          throw new IAE("second argument should be string type but got %s type", format.type());
-        }
-        formatter = DateTimeFormat.forPattern(format.asString());
-      }
-      DateTime date;
-      try {
-        date = DateTime.parse(value.asString(), formatter);
-      }
-      catch (IllegalArgumentException e) {
-        throw new IAE(e, "invalid value %s", value.asString());
-      }
-      return toValue(date);
-    }
-
-    protected ExprEval toValue(DateTime date)
-    {
-      return ExprEval.of(date.getMillis());
-    }
-  }
-
-  class UnixTimestampFunc extends TimestampFromEpochFunc
-  {
-    @Override
-    public String name()
-    {
-      return "unix_timestamp";
-    }
-
-    @Override
-    protected final ExprEval toValue(DateTime date)
-    {
-      return ExprEval.of(date.getMillis() / 1000);
-    }
-  }
-
-  class NvlFunc implements Function
-  {
-    @Override
-    public String name()
-    {
-      return "nvl";
-    }
-
-    @Override
-    public ExprEval apply(List<Expr> args, Expr.ObjectBinding bindings)
-    {
-      if (args.size() != 2) {
-        throw new IAE("function 'nvl' needs 2 arguments");
-      }
-      final ExprEval eval = args.get(0).eval(bindings);
-      return eval.isNull() ? args.get(1).eval(bindings) : eval;
-    }
+  // implement this and register to jackson module to extend expression function (see ExamplesDruidModule)
+  interface Library {
```

Comment:
It's just a marker/wrapper class to find with jackson. wish there is a better way.